Datastore/tests/System/DatastoreMultipleDBTestCase.php

```diff
@@ -0,0 +1,121 @@
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Datastore\Tests\System;
+
+use Google\Auth\CredentialsLoader;
+use Google\Cloud\Core\Testing\System\DeletionQueue;
+use Google\Cloud\Datastore\DatastoreClient;
+use GuzzleHttp\Client;
+
+/**
+ * Datastore does not use the default deletion queue. Because of the way
+ * datastore entities are deleted, a local queue is required.
+ * Be sure to use `self::$localDeletionQueue` for all datastore entities
+ * created during Multiple DB tests.
+ */
+class DatastoreMultipleDbTestCase extends DatastoreTestCase
+{
+    const TEST_DB_NAME = 'gcloud_test_multidb';
+
+    protected static $restMultiDbClient;
+    protected static $grpcMultiDbClient;
+    protected static $returnInt64AsObjectMultiDbClient;
+    private static $hasSetUp = false;
+
+    public static function set_up_multi_db_before_class()
+    {
+        if (self::$hasSetUp) {
+            return;
+        }
+
+        if (!self::checkTestDbExists()) {
+            // TODO: do a POST call to create DB
+        }
+
+        $config = [
+            'keyFilePath' => getenv('GOOGLE_CLOUD_PHP_TESTS_KEY_PATH'),
+            'namespaceId' => uniqid(self::TESTING_PREFIX),
+            'databaseId' => self::TEST_DB_NAME,
+        ];
+
+        self::$restMultiDbClient = new DatastoreClient($config + [
+            'transport' => 'rest',
+        ]);
+        self::$grpcMultiDbClient = new DatastoreClient($config + [
+            'transport' => 'grpc',
+        ]);
+        self::$returnInt64AsObjectMultiDbClient = new DatastoreClient($config + [
+            'returnInt64AsObject' => true,
+        ]);
+        self::$hasSetUp = true;
+    }
+
+    public function multiDbClientProvider()
+    {
+        // self::set_up_before_class();
+        self::set_up_multi_db_before_class();
+
+        return [
+            'multiDbRestClient' => [self::$restMultiDbClient],
+            'multiDbGrpcClient' => [self::$grpcMultiDbClient],
+        ];
+    }
+
+    private static function checkTestDbExists()
+    {
+        $keyFileContents = json_decode(
+            file_get_contents(getenv('GOOGLE_CLOUD_PHP_TESTS_KEY_PATH')),
+            true
+        );
+        $fetcher = CredentialsLoader::makeCredentials(
+            ['https://www.googleapis.com/auth/datastore'],
+            $keyFileContents
+        );
+        $token = $fetcher->fetchAuthToken();
+        $project_id = getenv('GOOGLE_PROJECT_ID');
+
+        $client = new Client();
+        $res = $client->get(
+            sprintf(
+                'https://firestore.googleapis.com/v1/projects/%s/databases',
+                $project_id
+            ),
+            [
+                'headers' => [
+                    'Authorization' => 'Bearer ' . $token['access_token'],
+                    'Accept' => 'application/json',
+                ],
+            ]
+        );
+
+        $dbs = json_decode($res->getBody(), true);
```

Comment:
@bshaffer is there any easier way to call Firestore APIs to check whether multiple DS are created or not?
There's also a POST call that I need to do on this URL to create DB if not already done.

This needs a composer change, to add `google/auth` and doesn't seem like the best approach.

Reply:
Following up with David on it.